src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -155,7 +156,14 @@ private void addBinaryToPath(String hdfsBinaryPath) throws IOException {
     BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
     bufferedWriter.write(scriptContent);
     bufferedWriter.close();
-    Runtime.getRuntime().exec("chmod a+x " + pathEnvVarLocation);
+    ProcessBuilder processBuilder = new ProcessBuilder("chmod", "a+x", pathEnvVarLocation);
+    Process process = processBuilder.start();
+    int exitCode = process.waitFor();
+    if (exitCode != 0) {
+      log.fatal("Error creating the symbolic link to hdfs binary."
+          + "Failure running chmod a+x " + pathEnvVarLocation);
+      System.exit(1);
```

Comment:
I think these are fatal errors. In the case (non predistributed binaries) that we need to create the symbolic link, if this does not happen, this is a major problem and the we should exit.